cmd/pinniped-proxy/src/main.rs

```diff
@@ -17,17 +18,26 @@ async fn main() -> Result<()> {
     pretty_env_logger::init();
     let opt = cli::Options::from_args();
 
+    // Load the default certificate authority data on startup once.
+    let default_ca_data = fs::read_to_string(opt.default_ca_cert.clone())
+        .with_context(|| format!("unable to load default-ca-cert at {}", opt.default_ca_cert))?;
+
     // For every incoming connection, we make a new hyper `Service` to handle
     // all incoming HTTP requests on that connection. This is done by passing a
     // closure to the hyper `make_service_fn` which returns our custom `make_svc`
-    // function that can be used for each connection. 
+    // function that can be used for each connection.
     let make_svc = make_service_fn(|_conn| {
+        let default_ca_data = default_ca_data.clone();
```

Comment:
At first this can look a little odd - why are we ghosting the variable which is already available in scope? The reason is that this closure needs to be able to reference the ca data in an async block below, but the async block, being async, could live longer than the referenced var from the parent scope. So we clone it here so the value is owned inside this closure, and similarly below when defining the service fn closure so it owns its own copy of the value.

Reply:
Thanks for the explanation; it's interesting how in Rust one should be aware of this ownership stuff. But, in the end, it's worthwhile as you get thread-safety out of the box